sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEPropagateEmptyRelation.scala

```diff
@@ -82,6 +82,21 @@ object AQEPropagateEmptyRelation extends PropagateEmptyRelationBase {
     case _ => false
   }
 
+  // In AQE, query stage will be wrapped with LogicalQueryStage, if it's child is
+  // BroadcastQueryStage and join other side is empty relation like the follow plan:
+  // Join
+  //   :- LocalTableScan <empty>, [a#23]
+  //   +- LogicalQueryStage(_, BroadcastQueryStage)
+  // After AQEPropagateEmptyRelation, the plan will be
+  // Project
+  //   +- LogicalQueryStage(_, BroadcastQueryStage)
+  // Then after LogicalQueryStageStrategy, will only remain BroadcastQueryStage after project,
+  // the plan can't execute.
```

Comment:
We can simply say
```
// A broadcast query stage can't be executed without the join operator.
// TODO: we can return the original query plan before broadcast.
```

Reply:
how hard is it to return the original query plan? Seems not hard as we just need to add a new `def returnSingleJoinSide` function in the base class, and unwrap broadcast stage in the AQE rule.